src/main/scala/dpla/ingestion3/harvesters/oai/OaiResponseProcessor.scala

```diff
@@ -18,40 +19,59 @@ object OaiResponseProcessor {
   private[this] val logger = LogManager.getLogger("OaiHarvester")
 
   /**
-    * Iterates through the records in an OAI response.
-    * Returns a Sequence of OaiRecord
+    * Parses records from an OaiSource.
     *
-    * @param xml NodeSeq
-    *            The XML response from the OAI request
+    * @param source OaiSource
+    *               The single-page response to a single OAI query.
+    *
+    * @return OaiResponse
+    *         OaiRecordsPage - all the records appearing on the page OR
+    *         OaiError - an error incurred during the process of parsing the records.
     */
 
-    def getRecords(xml: NodeSeq, set: Option[OaiSet] = None): Seq[OaiRecord] = {
-      val records = xml \\ "OAI-PMH" \\ "record"
-      records.map(r => r.headOption match {
-        case Some(node) => {
-          new OaiRecord(getOaiIdentifier(node), node.toString, set)
+  def getRecords(source: OaiSource): OaiResponse = {
+
+    // Parse all records from XML
+    def getRecordsFromXml(xml: Node): Seq[OaiRecord] = {
+      val xmlRecords: NodeSeq = xml \\ "OAI-PMH" \\ "record"
+
+      xmlRecords.flatMap(record => {
```

Comment:
You don't need curly braces here, either, since `record.headOption match {}` is a single statement